src/transforms/deployment.ts

```diff
@@ -1,7 +1,79 @@
 import _ from "lodash";
-import { Context } from "probot/lib/context";
 import issueKeyParser from "jira-issue-key-parser";
-import { JiraDeploymentData } from "../interfaces/jira";
+import {JiraDeploymentData} from "../interfaces/jira";
+import {GitHubAPI} from "probot";
+import {WebhookPayloadDeploymentStatus} from "@octokit/webhooks";
+import {LoggerWithTarget} from "probot/lib/wrap-logger";
+import {Octokit} from "@octokit/rest";
+import {booleanFlag, BooleanFlags} from "../config/feature-flags";
+
+async function getLastSuccessfulDeployCommitSha(
+	owner: string,
+	repoName: string,
+	github: GitHubAPI,
+	deployments: Octokit.ReposListDeploymentsResponseItem[]
+): Promise<string> {
+
+	for (const deployment of deployments) {
+		// Get each deployment status for this environment so we can have their statuses' ids
+		const listDeploymentStatusResponse: Octokit.Response<Octokit.ReposListDeploymentStatusesResponse> = await github.repos.listDeploymentStatuses({
+			owner: owner,
+			repo: repoName,
+			deployment_id: deployment.id
+		});
+
+		// Filter only the successful ones
+		const lastSuccessful: Octokit.ReposListDeploymentStatusesResponseItem[] = listDeploymentStatusResponse.data.filter(deployment => deployment.state === "success");
+		if (lastSuccessful) {
+			return deployment.sha;
+		}
+	}
+
+	// If there's no successful deployment on the list of deployments that GitHub returned us (max. 250) then we'll return the last one from the array, even if it's a failed one.
+	return deployments[deployments.length - 1].sha;
```

Comment:
why the last one?  Wouldn't the first one be the 'latest'?  If not, I'd recommend you set the order when you call the API since you're for loop is wrong in that case.

Reply:
Yes, the first one is the latest but if we return the latest when we do the commits diff, later on, we won't have any commits in between deployments - because they'll be the same. Grabbing the last of the array means grabbing the oldest one. AFAIK, the API returns deployments from newest to oldest.